marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -201,25 +201,30 @@ declare %private function hent:fix-options-for-explorer(
           hent:fix-options-for-explorer($n/node(), $sortable-properties, $entity-namespace-map)
         }
       case element(search:constraint) return
-        element { fn:node-name($n) } {
-          $n/@*,
-          let $path-expression := fix-path-expression(fn:string($n/search:range/search:path-index))
-          let $search-range-node := $n/search:range
-          let $is-sortable-only :=
-            let $sort-info := map:get($sortable-properties, $path-expression)
-            return
-              if (fn:exists($sort-info)) then map:get($sort-info, "is-sortable-only") = fn:true()
-              else fn:false()
-          return
-            if (fn:empty($search-range-node) or fn:not($is-sortable-only)) then
-              hent:fix-options-for-explorer($n/node(), $sortable-properties, $entity-namespace-map)
-            else
-              element {fn:node-name($search-range-node)} {
-                $search-range-node/attribute()[not(name()='facet')],
-                attribute facet {"false"},
-                hent:fix-options-for-explorer($search-range-node, $sortable-properties, $entity-namespace-map)/node()
-              }
-        }
+        let $search-container-node := $n/search:container
+        return
+          if (fn:not(fn:empty($search-container-node))) then
```

Comment:
Pinging @fsnow  to confirm, as I'm forgetting xquery each day, but I'm pretty certain that since you have a node or an empty sequence here, you can just do this:

    if ($search-container-node)

Though the more elegant approach would be this:

    let $container := $n/search:container
    where fn:not($container)
    return ...

I like let/where/return as a replacement for an if/else where either condition has empty sequence as a return value

Reply:
And to make it more self-documenting, you could do this:

let $container-for-entity-property-generated-by-es := $n/search:container

Reply:
Updated to let/where/return